public/assets/js/assetConstructor.js

```diff
@@ -0,0 +1,124 @@
+const Asset = function () {
+    this.state = {};
```

Comment:
I'd like to discuss this design and the purpose of it some in person. Please message me on chat and we'll get on a Hangout.